### ISSUE TITLE
Valid date literal not accepted in record literal

### ISSUE THREAD

**Issue by bobhy** [nushell/nushell#9186]
### Describe the bug

A full RFC3339 datetime literal is bad syntax within a record literal, but a simpler datetime works fine.
The full datetime literal works OK in a list, and the *value* of the literal works fine in a record literal.

### How to reproduce

```
〉2001-05-11T13:00:05    # a perfectly ordinary datetime
Fri, 11 May 2001 13:00:05 +0000 (22 years ago)
```
```
〉{a: 2001-05-11T13:00:05 }  # not accepted in record
Error: nu::parser::parse_mismatch

  × Parse mismatch during operation.
   ╭─[entry #3:1:1]
 1 │ {a: 2001-05-11T13:00:05 }
   · ────────────┬────────────
   ·             ╰── expected record
   ╰────
```
```
〉{a: 2001-05-11 } | get a | describe  # a simpler datetime literal is accepted as a datetime in a record literal
date
```
```
〉{ a : 2001-05-11T13:00:05 }    # various attempts at spacing make no difference
Error: nu::parser::parse_mismatch

  × Parse mismatch during operation.
   ╭─[entry #8:1:1]
 1 │ { a : 2001-05-11T13:00:05 }
   · ─────────────┬─────────────
   ·              ╰── expected record
   ╰────
```
```
〉{ a : 2001-05-11 }    # same spacing with simpler date literal works fine.
╭───┬──────────────╮
│ a │ 22 years ago │
╰───┴──────────────╯
```
```
〉[ 2001-05-11T13:00:05 ]  # failing literal works OK in a list
╭───┬──────────────╮
│ 0 │ 22 years ago │
╰───┴──────────────╯
```
```
〉let t = 2001-05-11T13:00:05  # seems to be a problem with the literal, not the value of the datetime
〉$t
Fri, 11 May 2001 13:00:05 +0000 (22 years ago)
〉{ a: $t }
╭───┬──────────────╮
│ a │ 22 years ago │
╰───┴──────────────╯
```

### Expected behavior

I expected any valid datetime literal to be usable as a value in a record (literal).

### Configuration

| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.79.1                                   |
| branch             | fix_9027                                 |
| commit_hash        | 271e4430fb3ee1fc7f744fd646d85c90c41b5e47 |
| build_os           | linux-x86_64                             |
| build_target       | x86_64-unknown-linux-gnu                 |
| rust_version       | rustc 1.67.1 (d5a82bbd2 2023-02-07)      |
| rust_channel       | 1.67.1-x86_64-unknown-linux-gnu          |
| cargo_version      | cargo 1.67.1 (8ecd4f20a 2023-01-10)      |
| build_time         | 2023-05-11 20:18:40 -04:00               |
| build_rust_channel | release                                  |
| features           | default, zip                             |
| installed_plugins  |                                          |

**Comment by amtoine** [comment]
looks like a parser thing :thinking: 

i suspect the `:` are responsible for this :open_mouth: 
```
{a: (2001-05-11T13:00:05)}
```
forces the date to evaluate to a `datetime` and works :+1:

**Comment by fdncred** [comment]
I agree, you'd think this would parse. Looks like a bug to me.

**Comment by bgeron** [comment]
This currently breaks NUON: `{a: (date now)} | to nuon | from nuon` is a parse error in `from nuon`.

**Comment by 1Kinoti** [comment]
the issue here is the `:`, which makes the parser think that this is another record entry

```nushell
{a: 2001-05-11T13:00:05 }
 ^^^^^^^^^^^^^^^ ^
        |        |
  first entry    |
                 expected a key but found the `:` separator
                 the parser thinks that you wanted to start
                 entry but forgor the key
```

we should either make the parser a little greedy here, backtracking if necessary. 

or, fix this [FIXME](https://github.com/nushell/nushell/blob/17caa3132577031d9b3ead8e481e9cab204d41b2/crates/nu-parser/src/parser.rs#L5383) to provide an better error message

**Comment by fdncred** [comment]
seems like we need better RFC3339/ISO8601 detector and parser maybe https://ijmacd.github.io/rfc3339-iso8601/

**Comment by weirdan** [comment]
I believe this has been fixed in #13413:

```console
$ {a: 2001-05-11T13:00:05 }
╭───┬──────────────╮
│ a │ 23 years ago │
╰───┴──────────────╯
```

**Comment by fdncred** [comment]
Good catch. Thanks!
